include/envoy/server/filter_config.h

```diff
@@ -0,0 +1,192 @@
+#pragma once
+
+#include <functional>
+
+#include "envoy/access_log/access_log.h"
+#include "envoy/http/filter.h"
+#include "envoy/init/init.h"
+#include "envoy/json/json_object.h"
+#include "envoy/network/filter.h"
+#include "envoy/ratelimit/ratelimit.h"
+#include "envoy/runtime/runtime.h"
+#include "envoy/server/drain_manager.h"
+#include "envoy/server/instance.h" // TODO(mattklein123): Remove post 1.4.0
+#include "envoy/tracing/http_tracer.h"
+#include "envoy/upstream/cluster_manager.h"
+
+namespace Envoy {
+namespace Server {
+namespace Configuration {
+
+/**
+ * Context passed to network and HTTP filters to access server resources.
+ * TODO(mattklein123): When we lock down visibility of the rest of the code, filters should only
+ * access the rest of the server via interfaces exposed here.
+ */
+class FactoryContext {
+public:
+  virtual ~FactoryContext() {}
+
+  /**
+   * @return AccessLogManager for use by the entire server.
+   */
+  virtual AccessLog::AccessLogManager& accessLogManager() PURE;
+
+  /**
+   * @return Upstream::ClusterManager& singleton for use by the entire server.
+   */
+  virtual Upstream::ClusterManager& clusterManager() PURE;
```

Comment:
I assume all of these are needed for existing filter and there is no further way to minimize the exposed surface.